src/App.tsx

```diff
@@ -1,17 +1,129 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import './App.scss';
+import { Todos } from './types/Todos';
+import { TodoList } from './components/TodoList/TodoList';
 
+import todos from './api/todos';
 import users from './api/users';
 
 const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState('');
+  const [todo, setTodo] = useState(todos);
+  const [status, setStatus] = useState('not completed');
+  const [errUser, setErrUser] = useState('');
+  const [errTitle, setErrTitle] = useState('');
+
+  const validate = () => {
+    if (!user) {
+      setErrUser('Please choose a user');
+    }
+
+    if (!title.trim()) {
+      setErrTitle('Please enter the title');
+    }
+
+    if (!user || !title.trim()) {
+      return false;
+    }
+
+    return true;
+  };
+
+  const addTodo = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+    if (validate()) {
+      const searchingUser = users.find(({ name }) => name === user);
+
+      const newTodo = {
+        id: todo[todo.length - 1].id + 1,
+        title,
+        userId: searchingUser ? searchingUser.id : 0,
+        completed: status === 'completed',
+      };
+
+      setTodo((current) => [...current, newTodo]);
+      setTitle('');
+      setUser('');
+      setStatus('not completed');
+    }
+  };
+
+  const preparedTodos: Todos[] = todo.map((oneTodo) => ({
+    ...oneTodo,
+    user: users.find((oneUser) => oneUser.id === oneTodo.userId),
+  }));
+
   return (
     <div className="App">
-      <h1>Add todo form</h1>
+      <h1 className="App__title">Static list of todos</h1>
+
+      <TodoList todos={preparedTodos} />
+
+      <form onSubmit={addTodo} className="App__form">
+        <h2 className="App__formTitle">Create a new todo</h2>
+
+        <div className="App__wrapper">
+          <label>
+            <input
+              type="text"
+              name="title"
+              value={title}
+              placeholder="Title"
+              onChange={event => {
+                const newTitle = event.target.value;
+
+                setTitle(newTitle.replace(/[^a-zA-Z0-9А-Яа-я\s]/g, ''));
+                setErrTitle('');
+              }}
+              className="App__inputTitle"
+            />
+          </label>
+          <span style={{ color: 'red' }}>{errTitle}</span>
```

Comment:
use scss file

Reply:
you don’t need to render an empty span if you don’t have errTitle

<img width="790" alt="Screenshot 2022-05-12 at 13 07 35" src="https://user-images.githubusercontent.com/47606379/168046920-dbdb67cd-1f03-4e9a-a5a6-7a75f163498b.png">